corfudb-tools/src/main/java/org/corfudb/browser/CorfuStoreBrowserEditor.java

```diff
@@ -188,15 +194,47 @@ private void printPayload(Map.Entry<CorfuDynamicKey, CorfuDynamicRecord> entry)
             log.error("payload is NULL");
             return;
         }
+
         try {
             builder = new StringBuilder("\nPayload:\n")
                     .append(JsonFormat.printer().print(entry.getValue().getPayload()));
-            log.info(builder.toString());
+            System.out.println(builder.toString());
         } catch (Exception e) {
             log.error("invalid payload: ", e);
         }
     }
 
+    private int printTableRegistry() {
+        for (Map.Entry<TableName, CorfuRecord<TableDescriptors, TableMetadata>> entry :
+                dynamicProtobufSerializer.getCachedRegistryTable().entrySet()) {
+            try {
+                StringBuilder builder = new StringBuilder("\nKey:\n")
+                        .append(JsonFormat.printer().print(entry.getKey()));
+                System.out.println(builder.toString());
+            } catch (Exception e) {
+                log.error("Unable to print metadata entry of this registry table key {}",  entry.getKey());
+            }
+            StringBuilder builder = new StringBuilder("\nkeyType = \"");
+            builder.append("\nkeyType = \""+entry.getValue().getPayload().getKey().getTypeUrl()+"\"");
+            builder.append("\npayloadType = \""+entry.getValue().getPayload().getValue().getTypeUrl()+"\"");
+            builder.append("\nmetadataType = \""+entry.getValue().getPayload().getMetadata().getTypeUrl()+"\"");
+            builder.append("\nProtobuf Source Files: \""+
+                    entry.getValue().getPayload().getFileDescriptorsMap().keySet()
+            );
+            System.out.println(builder);
```

Comment:
I think it should be `System.out.println(builder.toString());`
Also, can we wrap this part in a try-catch block too? The Key, Value, Metadata fields are populated during migration. If the migration failed and we want to inspect the table then this code could hit NPE.

Reply:
Thanks George will wrap it in a try-catch block too, even though it is an internal table. As for the toString, if I add the toString() the IDE tells me that it is redundant and makes me remove it!